Aquarius/Src/Aquarius/Core/Window.cpp

```diff
@@ -0,0 +1,109 @@
+#include "Window.h"
+#include "Log.h"
+#include <assert.h>
+
+namespace Aquarius {
+
+    Window::Window(uint32_t width, uint32_t height, std::string &&name, bool vsync)
+        : m_Width(width), m_Height(height), m_Name(std::move(name)), m_VsyncEnabled(vsync), m_Window(nullptr), m_Context(nullptr)
+    {}
+
+    Window::WindowPtr Window::Create(uint32_t width, uint32_t height, std::string &&name, bool vsync)
```

Comment:
I like wrapping the std::unique_ptr here, looks a lot cleaner

Reply:
Yeah std::unique_ptr<RenderingContext>and the like is pretty long and annoying